.github/workflows/ci.yml

```diff
@@ -54,6 +54,23 @@ jobs:
         run: cargo --version
       - run: cargo test --verbose --workspace --features diagnostics
 
+  clippy:
+    name: Lint with clippy
+    runs-on: ubuntu-latest
+    env:
+      RUSTFLAGS: -Dwarnings
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
```

Comment:
Why nightly here? Wouldn't `stable` produce more CI-friendly output?

Reply:
If Clippy runs with `--all-features`, it seems that nightly must be used or the `diagnostics` feature will not compile.

If you want, Clippy can also run without `--all-features` in stable, but in turn it can't lint against `diagnostics` code.

What do you mean with CI-friendly output, by the way?

Reply:
> or the `diagnostics` feature will not compile.

Sadly, I don't see that stabilizing anytime soon in rustc. It's probably more important to lint the case without that feature enabled.

> What do you mean with CI-friendly output, by the way?

I've been burned by nightly having changes that caused builds to fail for "no reason." Forget I said anything 😄

Reply:
Okay, the Clippy install is now changed back to stable and runs on default features